src/main/java/com/danielflower/apprunner/runners/GoRunner.java

```diff
@@ -0,0 +1,70 @@
+package com.danielflower.apprunner.runners;
+
+import com.danielflower.apprunner.problems.ProjectCannotStartException;
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.shared.invoker.InvocationOutputHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class GoRunner implements AppRunner {
+    public static final Logger log = LoggerFactory.getLogger(GoRunner.class);
+
+    public static final String[] startCommands = new String[]{"go build", "go test", "./{app_dir_name}"};
+    private final File projectRoot;
+    private final String goPath;
+    private final CommandLineProvider goCmd;
+    private ExecuteWatchdog watchDog;
+
+    public GoRunner(File projectRoot, String goPath, CommandLineProvider goCmd) {
+        this.projectRoot = projectRoot;
+        this.goPath = goPath;
+        this.goCmd = goCmd;
+    }
+
+    @Override
+    public File getInstanceDir() {
+        return projectRoot;
+    }
+
+    public void start(InvocationOutputHandler buildLogHandler, InvocationOutputHandler consoleLogHandler, Map<String, String> envVarsForApp, Waiter startupWaiter) throws ProjectCannotStartException {
+        envVarsForApp.put("GOPATH", goPath);
+        try {
+            rungo(buildLogHandler, envVarsForApp, "get");
```

Comment:
Firstly, go doesn't recommend to have independent workspace for each app. GoLang want all go project to be in one GOPATH so that it has more centralized way to manage the package, src and binaries.

While we just want to run the app rather than to develop it, it is much easier to change in this way. But maybe it is better to have isolated workspace for each app, we can discuss that.